In [2]:
import scipy
from scipy import stats
from scipy.stats import norm
from scipy.stats import t

In [2]:
norm.cdf(-1)

0.15865525393145707

# norm.cdf() - input

- X axis value
- It can be negative

# norm.cdf() - output

- area under the curve on the left side of the value that is given as the input
- output is always positive, area is cannot be less than zero

# ppf() - probability point function

In [4]:
norm.ppf(0.15865525393145707)

-1.0

In [4]:
norm.ppf(0.975)

1.959963984540054

In [10]:
norm.ppf(0.5)

0.0

# norm.ppf() - input

- area under the curve on left side.
- area cannot be negative.

# norm.ppf() - output

- value on the X axis.
- It can be negative.

In [5]:
norm.ppf(0.025)

-1.9599639845400545

In [6]:
norm.ppf(0.975)

1.959963984540054

In [7]:
norm.ppf(0.5)

0.0

1. Establish null and alternative hypothesis
Null hypothesis
- is made in such a way that of it is found to be true after analysis
- no action
- no design
- no changes

# Continuos data tools

- 1 sample z test or 1 sample t test
- compare the mean of a single sample with the claim value and establish whether the difference is statistically significant or not.

In [8]:
x = [4,7,6,8,9,7,2,3,5,6,7,8]

In [9]:
import numpy as np
import pandas as pd

In [12]:
import statsmodels

In [13]:
from statsmodels import stats

In [14]:
from statsmodels.stats import weightstats

In [15]:
np.mean(x)

6.0

Ho avg(x) is atleast 7.2 Ho a>=7.2 Ho a<7.2

CL = 95%, LOS = 5% = 0.05
- choose 1 sample 1 test or z test to compare a sample mean with the claim value

# Sequence of input

- 1. list name
- 2. claim value
- 3. sign of alternative hyphothesis

In [16]:
statsmodels.stats.weightstats.ztest(x,value=7.2,alternative='smaller')

(-1.9497692171126306, 0.025601816055026008)

# 'p-' value is less than LOS, then action is required.

- if p-value is less than 0.05 so we reject the Ho. 
The difference between 6 and 7.2 is statistically significant, cannot be ignoed so the action is required.

In [17]:
statsmodels.stats.weightstats.ztest(x,value=6.5,alternative='smaller')

(-0.812403840463596, 0.2082799716383955)

- p-value is more than 0.05 so we do not reject the Ho. The difference between 6 and 6.5 is statistically NOT significant, can be ignored so the action is not required.
- 21% chance of commiting type 1 error
- p-value > LOS(0.05) we do not reject Ho